lib/dom/genDomScripts.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * We have a collection of typescript functions that we need to run in the browser.
+ * First, we build them into a single js file
+ * Second, due to framework differences we need to get our script content as a string to avoid pathing issues due to file routing in frameworks like Next.js
+ * Playwright allows us to pass in script content directly as a string instead of reading a file from a path
+ * https://github.com/browserbase/stagehand/issues/180
+ *
+ * We can't rely on the normal build process for stagehand, because we need our script content as a string so that the import *just works*
+ */
+const fs = require("fs");
+const path = require("path");
+
+const esbuild = require("esbuild");
+
+fs.mkdirSync(path.join(__dirname, "./build"), { recursive: true });
+
+esbuild.buildSync({
+  entryPoints: [path.join(__dirname, "index.ts")],
+  bundle: true,
+  outdir: path.join(__dirname, "build"),
+});
+
+const scriptContent = fs.readFileSync(
+  path.join(__dirname, "./build/index.js"),
+  "utf8",
+);
+
+const output = `export const scriptContent = ${JSON.stringify(scriptContent)};`;
```

Comment:
is this kosher? seems scary to have stringified js

Reply:
it's not super kosher! but if you think about it the only way to get this script into a remote browser is to send it across the "wire" as data anyways right?

Reply:
fair haha, just wanted to see if there was a way to have stuff like syntax highlighting. if this is the only way then ig we have no other option

Reply:
yeah luckily the logic is self is typescript, it's just this bridge to a string that we have less control over